src/ui/public/filter_editor/filter_editor.js

```diff
@@ -35,7 +35,13 @@ module.directive('filterEditor', function ($timeout, indexPatterns) {
     controllerAs: 'filterEditor',
     bindToController: true,
     controller: callAfterBindingsWorkaround(function ($scope, $element) {
-      this.init = () => {
+      this.init = async () => {
+        if (this.indexPatterns === undefined || this.indexPatterns.length === 0) {
```

Comment:
Hmm, looks like sometimes `this.indexPatterns` is `[undefined]` (like when you go to visualize and create a new timelion vis), so you'll have to check for that condition as well.